org.omg.sysml/src/org/omg/sysml/util/repository/EMFModelRefresher.java

```diff
@@ -0,0 +1,257 @@
+	}
+
+	private void transformContainmentReferences(EObject langElement, Element remoteElement) {
+		for (EStructuralFeature feature: getStructuralFeatuers(langElement))
+		{
+			if (!feature.isDerived() && remoteElement.get(feature.getName()) != null) {
+				if (feature instanceof EReference reference && reference.isContainment()) {
+					transformStructuralFeature(langElement, remoteElement, reference, true, true);
+				}
+			}
+		}
+	}
+	
+	private void tranformCrossreferences(EObject langElement, Element remoteElement) {
+		for (EStructuralFeature feature: getStructuralFeatuers(langElement))
+		{
+			if (!feature.isDerived() && remoteElement.containsKey(feature.getName())) {
+				if (feature instanceof EReference reference) {
+					transformStructuralFeature(langElement, remoteElement, reference, true, false);
+				}
+			}
+		}
+		
+		if (langElement instanceof FeatureTyping ft) {
+			ft.setTypedFeature(null);
+		}
+	}
+	
+	private void tranformAttributes(EObject langElement, Element remoteElement) {
+		for (EStructuralFeature feature: getStructuralFeatuers(langElement))
+		{
+			if (!feature.isDerived() && remoteElement.containsKey(feature.getName())) {
+				if (feature instanceof EAttribute reference) {
+					transformStructuralFeature(langElement, remoteElement, reference, false, false);
+				}
+			}
+		}
+	}
+	
+	private static Collection<EStructuralFeature> getStructuralFeatuers(EObject langElement){
+		EClass eClass = langElement.eClass();
+		
+		Set<EObject> allRedefined = eClass.getEAllStructuralFeatures().stream()
+				.map(f -> f.getEAnnotation("redefines")).filter(Objects::nonNull)
+				.flatMap(a -> a.getReferences().stream()).collect(Collectors.toSet());
+		
+		return eClass.getEAllStructuralFeatures().stream().filter(f -> !allRedefined.contains(f)).collect(Collectors.toList());
+	}
+	
+	@SuppressWarnings("unchecked")
+	private void transformStructuralFeature(EObject langElement,  Element remoteElement, EStructuralFeature feature, boolean isReference, boolean isContainment) {
+		if (!disabledStructuralFeatures.contains(feature.getName())) {
+			try {
+				if (feature.isMany()) {
+					var referenceList = (List<EObject>) langElement.eGet(feature, false);
+					Object value = remoteElement.get(feature.getName());
+					if (value instanceof Collection valueCollection) {
+						for (Object referenceValue: valueCollection) {
+							EObject referencedLangElement = resolveReference(referenceValue, isContainment);
+							if (referencedLangElement != null) {
+								referenceList.add(referencedLangElement);
+							}
+						}
+					} 
+				} else {
+					Object value = remoteElement.get(feature.getName());
+					if (value != null) {
+						langElement.eSet(feature, isReference? resolveReference(value, isContainment) : transformAttributeValue(value, feature));
+					}
+				}
+			}
+			catch (Exception e) {
+				System.out.printf("Unable to set structural feature %s::%s %n", feature.getEContainingClass().getName(), feature.getName());

```

Comment:
Is there no reasonable alternative to error handling? This basically ignores the fact that an exception was thrown (any kind!), but can write the results to the standard output of the application that is not really meaningful in an Eclipse or Jupyter environment.

Reply:
Handled in #616